### ISSUE TITLE
/en-US/docs/Web/API/RTCPeerConnection/icecandidate_event

### ISSUE THREAD

**Issue by bardiharborow** [mdn/sprints#2450]
# Request type

- [ ] New documentation
- [x] Correction or update

# Details
See [icecandidate event](https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/icecandidate_event).

The code sample in "Sharing a new candidate" checks if event.candidate is truthy before delivering it to the remote peer. "Indicating the end of a generation of candidates" says that events where `event.candidate == ""` should be delivered, but [the empty string is falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) in JavaScript and this signal would not be delivered. "Indicating that ICE gathering is complete" says that events where `event.complete == null` should not be delivered but then refers to the `event.candidate == true` test.

There is clearly a nuance here that I'm missing, but in any case this may need to be clarified. An uninformed reader would form the conclusion that actually a `event.complete != null` test is needed.

There's also mixed references to RTCPeerConnectionIceEvent (bluelinked) and RTCPeerIceCandidateEvent (redlinked), not sure if that's intentional.

**Comment by a2sheppy** [comment]
Yeah, the meaning of a null candidate changed a while back and the document clearly has not fully caught up. That needs to be cleared up. And yes, the references to RTCPeerIceCanddiateEvent are wrong and should be RTCPeerConnectionIceEvent.

**Comment by bardiharborow** [comment]
@chrisdavidmills the main issue here doesn't appear to have been resolved, though the incorrect reference to RTCPeerIceCanddiateEvent seems to be gone.

**Comment by chrisdavidmills** [comment]
@bardiharborow OK, thanks for the heads up. I'll reopen this and move it to our current content repo.
